### ISSUE TITLE
Py3 - Nitrogen - Using TCP transport gives traceback 

### ISSUE THREAD

**Issue by frogunder** [saltstack/salt#41407]
### Description of Issue/Question
Using the TCP transport in Nitrogen gives traceback.

### Setup
Set up a py3 setup with `transport: tcp`

### Steps to Reproduce Issue
Seeing this when running commands:
```
Exception ignored in: <generator object _stream_return at 0x7ff240fd1b48>
Traceback (most recent call last):
  File "/root/salt/salt/transport/tcp.py", line 893, in _stream_return
AttributeError: 'NoneType' object has no attribute 'iostream'
```
It is not showing up each time though:
```
root@localhost:~# salt min* test.ping
minion_B:
    True
minion_A:
    True
Exception ignored in: <generator object _stream_return at 0x7f598dafbb48>
Traceback (most recent call last):
  File "/root/salt/salt/transport/tcp.py", line 893, in _stream_return
AttributeError: 'NoneType' object has no attribute 'iostream'
root@localhost:~# salt min* test.ping
minion_B:
    True
minion_A:
    True
Exception ignored in: <generator object _stream_return at 0x7ff240fd1b48>
Traceback (most recent call last):
  File "/root/salt/salt/transport/tcp.py", line 893, in _stream_return
AttributeError: 'NoneType' object has no attribute 'iostream'
root@localhost:~# salt min* test.ping
minion_B:
    True
minion_A:
    True
root@localhost:~# salt min* test.ping
minion_B:
    True
minion_A:
    True
root@localhost:~# salt min* test.ping
minion_B:
    True
minion_A:
    True
root@localhost:~# salt min* test.ping
minion_B:
    True
minion_A:
    True
Exception ignored in: <generator object _stream_return at 0x7fcac07beb48>
Traceback (most recent call last):
  File "/root/salt/salt/transport/tcp.py", line 893, in _stream_return
AttributeError: 'NoneType' object has no attribute 'StreamClosedError'
root@localhost:~# salt min* test.ping
minion_B:
    True
minion_A:
    True
Exception ignored in: <generator object _stream_return at 0x7f45d7e42b48>
Traceback (most recent call last):
  File "/root/salt/salt/transport/tcp.py", line 893, in _stream_return
AttributeError: 'NoneType' object has no attribute 'iostream'
root@localhost:~# salt min* test.ping
minion_B:
    True
minion_A:
    True
Exception ignored in: <generator object _stream_return at 0x7f660a8b9b48>
Traceback (most recent call last):
  File "/root/salt/salt/transport/tcp.py", line 893, in _stream_return
TypeError: catching classes that do not inherit from BaseException is not allowed
root@localhost:~# salt min* test.ping
minion_B:
    True
minion_A:
    True
Exception ignored in: <generator object _stream_return at 0x7f203f17ab48>
Traceback (most recent call last):
  File "/root/salt/salt/transport/tcp.py", line 893, in _stream_return
AttributeError: 'NoneType' object has no attribute 'iostream'
root@localhost:~# salt min* test.ping
minion_B:
    True
minion_A:
    True
root@localhost:~# salt min* test.ping
minion_B:
    True
minion_A:
    True
Exception ignored in: <generator object _stream_return at 0x7fd1e2dc6b48>
Traceback (most recent call last):
  File "/root/salt/salt/transport/tcp.py", line 893, in _stream_return
AttributeError: 'NoneType' object has no attribute 'iostream'
root@localhost:~# 
```
### Versions Report
```
Salt Version:
           Salt: 2017.5.0-97-gdb109a3
 
Dependency Versions:
           cffi: Not Installed
       cherrypy: Not Installed
       dateutil: Not Installed
      docker-py: Not Installed
          gitdb: Not Installed
      gitpython: Not Installed
          ioflo: Not Installed
         Jinja2: 2.9.6
        libgit2: Not Installed
        libnacl: Not Installed
       M2Crypto: Not Installed
           Mako: Not Installed
   msgpack-pure: Not Installed
 msgpack-python: 0.4.8
   mysql-python: Not Installed
      pycparser: Not Installed
       pycrypto: 2.6.1
   pycryptodome: Not Installed
         pygit2: Not Installed
         Python: 3.5.2 (default, Nov 17 2016, 17:05:23)
   python-gnupg: Not Installed
         PyYAML: 3.12
          PyZMQ: 16.0.2
           RAET: Not Installed
          smmap: Not Installed
        timelib: Not Installed
        Tornado: 4.5.1
            ZMQ: 4.1.6
 
System Versions:
           dist: Ubuntu 16.04 xenial
        machine: x86_64
        release: 4.9.15-x86_64-linode81
         system: Linux
        version: Ubuntu 16.04 denial
```

**Comment by garethgreenaway** [comment]
@frogunder So far I haven't been able to duplicate this issue.  I started a brand new Ubuntu vagrant instance, installed 2017.5 via source, configured the transport to be TCP and ran the tests using test.ping but I didn't see the errors.

My version report is a bit different, mostly tornado and zmq.
```
Salt Version:
           Salt: 2017.5.0-204-g8188ecf
 
Dependency Versions:
           cffi: Not Installed
       cherrypy: Not Installed
       dateutil: Not Installed
      docker-py: Not Installed
          gitdb: Not Installed
      gitpython: Not Installed
          ioflo: Not Installed
         Jinja2: 2.8
        libgit2: Not Installed
        libnacl: Not Installed
       M2Crypto: Not Installed
           Mako: Not Installed
   msgpack-pure: Not Installed
 msgpack-python: 0.4.6
   mysql-python: Not Installed
      pycparser: Not Installed
       pycrypto: 2.6.1
   pycryptodome: Not Installed
         pygit2: Not Installed
         Python: 3.5.2 (default, Nov 17 2016, 17:05:23)
   python-gnupg: Not Installed
         PyYAML: 3.11
          PyZMQ: 15.2.0
           RAET: Not Installed
          smmap: Not Installed
        timelib: Not Installed
        Tornado: 4.2.1
            ZMQ: 4.1.4
 
System Versions:
           dist: Ubuntu 16.04 xenial
        machine: x86_64
        release: 4.4.0-78-generic
         system: Linux
        version: Ubuntu 16.04 xenial
```

**Comment by garethgreenaway** [comment]
Did some more digging and it looks like the issues are related to changes made in tornado 4.5.  Installing tornado 4.4.3 and earlier versions do not have the issue.

**Comment by garethgreenaway** [comment]
Looks like #41436 should fix this.

**Comment by cachedout** [comment]
Fixed via the merge of #41436
